### ISSUE TITLE
Improve header(h1/h2/h3) styling in hover markdown

### ISSUE THREAD

**Issue by sam-mccall** [microsoft/vscode#88417]
Our LSP server produces markdown for hover, and the way headers are rendered doesn't seem ideal. Two main issues:

# Font sizes are too large

This natural-looking markdown:

`````markdown
# function `x`

---

`void x();`
`````

renders with an extremely large title that's inconsistent with overall VSCode font sizes.

![image](https://user-images.githubusercontent.com/548993/72149988-bd314f80-33a4-11ea-816e-f032e4837137.png)

There's a fairly easy workaround of using `<h3>` (`###`) instead, but this isn't ideal for language servers that are supposed to be portable across editors.

# Too much spacing

There's a lot of value in popups being unobtrusive, and the default margin of 1em h1/h2/h3 is very high for such a context.

For the following markdown:
`````markdown
### function `x`

---

`void x();`
`````
The current rendering is margin-heavy:
![image](https://user-images.githubusercontent.com/548993/72150443-d5ee3500-33a5-11ea-97a2-4178d6e6d8dd.png)

With top/bottom margin reduced to `0.5em`, IMO this is at least as easy to read:
![image](https://user-images.githubusercontent.com/548993/72150375-b525df80-33a5-11ea-82ed-33be95c9188b.png)

**Comment by octref** [comment]
I agree it doesn't look nice, but if not for a different size, how do you distinguish it from `**bold**`? And how do you tell h1/h2/h3 apart?

One idea I have would be to render h1/h2/h3 in bold with the leading `#` to indicate h level (idea from iA Writer):

![image](https://user-images.githubusercontent.com/4033249/72201645-9f242780-3456-11ea-90bc-0399add9333b.png)

But that would be a drastic change that needs to be discussed. I'm open to other ideas as well.

**Comment by panoply** [comment]
Personally, I feel like the default font sizing of hovers (tooltips) seems aggressively large for headers and descriptives. I was recently working with the `onHover` and `onCompletionResolve` handlers in an LSP implementation and began to notice this. Of course these capabilities are rendering in association with `editor.fontSize` and `editor.fontFamily` preferences set by the user.

<img width="661" alt="Screenshot 2020-02-06 at 20 20 18" src="https://user-images.githubusercontent.com/7041324/73971412-1eeec600-491f-11ea-955b-0b58d633a813.png">

An option to override the default presets markdown content is rendered with for font sizes and heading padding/margin would be useful for many implementations leveraging such capabilities. Even if said options had limitations in place or were predefined (xs, sm, md, lg, xl etc etc).

**Comment by Tyriar** [comment]
Current situation for workbench hovers:

<img width="116" alt="Screenshot 2023-05-28 at 9 38 26 am" src="https://github.com/microsoft/vscode/assets/2193314/ef599387-4ee9-48b5-817e-868fc2382ffb">

**Comment by Tyriar** [comment]
They're all getting mapped to h1s for some reason 😕 

<img width="242" alt="Screenshot 2023-05-28 at 9 43 05 am" src="https://github.com/microsoft/vscode/assets/2193314/40617f7a-37ee-4a4f-8a9f-e4ad1bece1cb">

**Comment by yilei** [comment]
The spacing seems is improved 🎊

Were the h1/h2/h3 header font sizes changed? They still look too large to me (and remain the same as OP):

<img width="456" alt="Screenshot 2023-05-30 at 17 40 55" src="https://github.com/microsoft/vscode/assets/581947/82ab97d4-b981-44d2-abc6-7f3b476bf773">

**Comment by Tyriar** [comment]
@yilei h1 is certainly much smaller now in your screenshot compared to OP. I'll leave this to @daviddossett (on leave atm) on whether we should make any further refinements. @daviddossett maybe we should make h4/5 the same as regular bold and shift the others down?

**Comment by yilei** [comment]
I tried to scale the two images so that their body font matches in size for a good comparison, and indeed h1 is smaller now, though not too much:

![Frame 46](https://github.com/microsoft/vscode/assets/581947/61028081-e1a8-43f6-b444-85f4c4de52f5)

Thanks for reopening!
